src/Search/Signals.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace eLife\Search;
+
+class Signals
+{
+    private static $isValid = true;
```

Comment:
As for previous comments, an object with a Factory Method in Kernel and injected in the right places would be more amenable to polymorphism